firefox-ios/Client/Coordinators/Library/LibraryCoordinator.swift

```diff
@@ -135,12 +135,55 @@ class LibraryCoordinator: BaseCoordinator,
         if isBookmarkRefactorEnabled {
             (navigationController.topViewController as? BookmarksViewController)?
                 .bookmarkCoordinatorDelegate = bookmarksCoordinator
+            if let recentFolderGUID = profile.prefs.stringForKey(PrefsKeys.LastViewedBookmarkFolder) {
+                CATransaction.begin()
+                CATransaction.setDisableActions(true)
+
+                populateBookmarksNavigationController(currentFolderGUID: recentFolderGUID,
+                                                      bookmarksCoordinator: bookmarksCoordinator) {
+                        CATransaction.commit()
+                }
+            }
         } else {
             (navigationController.topViewController as? LegacyBookmarksPanel)?
                 .bookmarkCoordinatorDelegate = bookmarksCoordinator
         }
     }
 
+    private func populateBookmarksNavigationController(currentFolderGUID: String,
+                                                       bookmarksCoordinator: BookmarksCoordinator,
+                                                       completion: @escaping () -> Void) {
+        profile.places.getBookmarksTree(rootGUID: BookmarkRoots.MobileFolderGUID,
+                                        recursive: true).uponQueue(.main) { result in
+            guard let maybeBookmarkTreeRoot = result.successValue,
+                  let bookmarkTreeRoot = maybeBookmarkTreeRoot else { return }
+            guard var path = self.findPathToFolder(targetGUID: currentFolderGUID,
+                                                   treeRootNode: bookmarkTreeRoot),
+                  !path.isEmpty else {return}
+            path.removeFirst()
+            path.forEach { el in
```

Comment:
What does `el` stands for? Can we have another name for this variable?

Reply:
Also, when this `forEach` is called, what does it look like if we have a lot of nested folders? Animations are set to false, so I guess to the user it's as if we magically appeared inside the nested folder?

Reply:
Yep, the behaviour is the same as if we were to close the library panel, and re-open it without restarting the app. From what I understood this was the behaviour we were trying to mimic. If we would like to have animations active I can do that as well.

Reply:
Without animation is good! Just wanted to confirm I understood properly, thank you!